cardano-api/src/Cardano/Api/Value.hs

```diff
@@ -406,6 +417,9 @@ renderPolicyId (PolicyId scriptHash) = serialiseToRawBytesHexText scriptHash
 
 renderAssetId :: AssetId -> Text
 renderAssetId AdaAssetId = "lovelace"
-renderAssetId (AssetId polId (AssetName assetName))
-  | BS.null assetName = renderPolicyId polId
-  | otherwise         = renderPolicyId polId <> "." <> Text.decodeUtf8 assetName
+renderAssetId (AssetId polId (AssetName "")) = renderPolicyId polId
+renderAssetId (AssetId polId assetName) =
+  renderPolicyId polId <> "." <> renderAssetName assetName
+
+renderAssetName :: AssetName -> Text
```

Comment:
`renderAssetName` is overkill. We just use `serialiseToRawBytesHexText`.

Reply:
done